### PR TITLE
Add ONNX export for BeiT

### PR DESCRIPTION
**Pull request by akuma12**
# What does this PR do?

Based on the fantastic work by @lewtun in https://github.com/huggingface/transformers/pull/15658, I added ONNX export support for the Microsoft BeiT model.

# Usage
```
import requests
import numpy as np
from PIL import Image
from onnxruntime import InferenceSession
from transformers import AutoConfig, AutoFeatureExtractor, AutoModelForImageClassification

# Export ViT checkpoint with image classification head
model_ckpt = "microsoft/beit-base-patch16-224"
!python -m transformers.onnx --model={model_ckpt} --feature=image-classification onnx/

# Download an image of two cute cats - naturally ;-)
url = "http://images.cocodataset.org/val2017/000000039769.jpg"
image = Image.open(requests.get(url, stream=True).raw)

# Instantiate config and feature extractor
config = AutoConfig.from_pretrained(model_ckpt)
feature_extractor = AutoFeatureExtractor.from_pretrained(model_ckpt)
inputs = feature_extractor(image, return_tensors="np")

# Create ONNX Runtime session
session = InferenceSession("onnx/model.onnx", providers=["CPUExecutionProvider"])
outputs = session.run(["logits"], dict(inputs))
predicted_class_idx = np.argmax(outputs[0])
# Returns Predicted class: Egyptian cat
print("Predicted class:", config.id2label[predicted_class_idx])
```

### CODE REVIEW

File: src/transformers/models/beit/configuration_beit.py [hunk 1]
```diff
@@ -176,3 +181,20 @@ def __init__(
         self.auxiliary_num_convs = auxiliary_num_convs
         self.auxiliary_concat_input = auxiliary_concat_input
         self.semantic_loss_ignore_index = semantic_loss_ignore_index
+
+
+class BeitOnnxConfig(OnnxConfig):
```
Comment (lewtun) [inline src/transformers/models/beit/configuration_beit.py:186]: I suggest adding the following comment above the class so we can cross-reference any changes on the ViT side :)

```suggestion
# Copied from transformers.models.vit.configuration_vit.ViTOnnxConfig
class BeitOnnxConfig(OnnxConfig):
```

**Review by lewtun — APPROVED** [review]
Thank you for this really clean contribution @akuma12 🚀 ! 

Out of curiosity, did you see much improvement when using dynamic quantization with BEIT? I found ViT only gave ~5-15%, so am curious if this is a general property of vision transformers ...

Gently pinging @sgugger or @LysandreJik for final approval 🤗

**Review by sgugger — APPROVED** [review]
Thanks for your PR! LGTM with @lewtun 's comment.

File: src/transformers/models/beit/configuration_beit.py (hunk 1 above)
Comment (akuma12) [reply src/transformers/models/beit/configuration_beit.py:187]: Thanks @lewtun! I added the recommended comment.

I haven't done any benchmarking of the original model vs ONNX, but I would be quite curious about that as well. Ultimately we're trying to use ONNX as a stepping stone to converting to TensorRT in the hopes of eking out even more performance, but if you have a benchmarking script that I could borrow I'd be happy to run it against BEiT and share the results. I checked the Transformers benchmarking utility and it doesn't look like it supports image classification models yet.

Comment (lewtun) [reply src/transformers/models/beit/configuration_beit.py:187]: Thanks for the update! We don't have a proper benchmarking script for ONNX models, but we should definitely think about including one in [`optimum`](https://github.com/huggingface/optimum) which is focused on optimisation topics. I've opened an issue here that you can track / comment on: https://github.com/huggingface/optimum/issues/128

**Comment by NielsRogge** [comment]
Thanks for your PR, super cool!

**Pull request merged**